hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/commit/SparkMergeHelper.java

```diff
@@ -77,14 +89,39 @@ public void runMerge(HoodieTable<T, JavaRDD<HoodieRecord<T>>, JavaRDD<HoodieKey>
       readSchema = mergeHandle.getWriterSchemaWithMetaFields();
     }
 
+    Option<InternalSchema> querySchemaOpt = SerDeHelper.fromJson(table.getConfig().getInternalSchema());
+    Boolean needToReWriteRecord = false;
+    // to do support bootstrap
+    if (querySchemaOpt.isPresent() && !baseFile.getBootstrapBaseFile().isPresent()) {
+      // check implicitly add columns, and position reorder(spark sql may change cols order)
+      InternalSchema querySchema = AvroSchemaUtil.evolutionSchemaFromNewAvroSchema(readSchema, querySchemaOpt.get(), true);
```

Comment:
Can we add a short-circuit to avoid performing these steps for EmptySchema (DummySchema) ? We need to make sure existing flow (when there is no schema evolution enabled) has minimal/no overhead not just in this place but everywhere

Reply:
Thanks

Reply:
fixed, yes, we use needToReWriteRecord to control the execution logic without affecting the original code logic